### PR TITLE
Docs: Document the Rust API

### PR DESCRIPTION
**Pull request by shashwatjolly**

### CODE REVIEW

File: src/service.rs [hunk 1]
```diff
@@ -462,25 +485,53 @@ impl CollectionManager {
...
+    /// # Arguments:
+    /// * `collection_type` - the type of [Item]s stored in the collection
+    /// * `meta` - the [ItemMetadata] for the collection  
+    /// * `content` - the collection's content as a byte array. This is unrelated to the [Item]s in the collection. 
     pub fn create<T: MsgPackSerilization>(&self, collection_type: &str, meta: &T, content: &[u8]) -> Result<Collection> {
         let meta = meta.to_msgpack()?;
         self.create_raw(collection_type, &meta, content)
     }
 
+    /// Create a new encrypted [Collection] and return a handle to it. This receives the metadata as a byte array.
```
Comment (tasn) [inline src/service.rs:499]: I think it's better to split it to two lines. Potentially with an empty line in between (like git) to better emphasise the first sentence as the main one.

```
/// Create a new encrypted [Collection] and return a handle to it using raw metadata
///
/// Unlike [create] metadata is expected to be a valid msgpack-encoded metadata object.
///
/// # Arguments:
/// ...

File: src/service.rs [hunk 2]
```diff
@@ -598,7 +710,11 @@ impl ItemManager {
         })
     }
 
-
```
Comment (tasn) [inline src/service.rs:601]: The extra empty line was on purpose.

File: src/service.rs [hunk 3]
```diff
@@ -640,6 +780,12 @@ impl ItemManager {
         self.item_manager_online.transaction(items, deps, options)
     }
 
+    /// Create an upload transaction for the supplied [Item]s with a list of items as dependencies. This will fail if the dependencies have changed remotely.
```
Comment (tasn) [inline src/service.rs:783]: Forgot to split this to two lines.

File: src/service.rs [hunk 4]
```diff
@@ -810,57 +994,81 @@ impl Collection {
...
     pub fn set_content(&mut self, content: &[u8]) -> Result<()> {
         self.col.set_content(&self.cm, content)
     }
 
+    /// Return the content of the collection as a byte array
     pub fn content(&self) -> Result<Vec<u8>> {
         self.col.content(&self.cm)
     }
 
+    /// Mark the collection as deleted
```
Comment (tasn) [inline src/service.rs:1023]: Maybe in the long description (for this, and for the one for item) mention that the item is only marked for deletion and still needs to be uploaded?

**Comment by tasn** [comment]
Note to self: document `meta_generic`

**Comment by shashwatjolly** [comment]
@tasn Documentation for `Client` is remaining

File: src/service.rs [hunk 5]
```diff
@@ -826,57 +1033,83 @@ impl Collection {
...
 
+    /// The type of the collection
     pub fn collection_type(&self) -> Result<String> {
         self.col.collection_type(&self.account_crypto_manager)
     }
 }
 
+/// The primary units of encrypted data in the application
+///
+/// Items can hold whatever kind of data is needed in the form of a byte array
```
Comment (tasn) [inline src/service.rs:1111]: Needed? Whatever?

How about replacing this and the next with:
```
Items have two pieces of data associated with them:
* [metadata](ItemMetadata) - contains meta information like name and modification time.
* Content - a buffer containing arbitrary binary data.
```

Comment (tasn) [reply src/service.rs:1111]: Same for collections if you did the same for that.

**Review by tasn — CHANGES_REQUESTED** [review]
Wow, that took me a while! Looks great overall, good job! I just made some suggestions.
The overall comment you need to keep in mind: the docs are here to simplify using the API, so they should:
1. Be concise, so they are easy to read.
2. Be explicit rather than vague
3. Simple, so don't add new terminology unless you have to

**Pull request merged**